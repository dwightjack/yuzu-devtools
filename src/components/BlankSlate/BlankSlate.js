import { html } from 'lit-html';
import { component } from 'haunted';

export default function blankSlate() {
  return html`
    <style>
      .blank {
        margin: 0;
        padding: var(--gutter);
        color: var(--color-quiet);
        font-size: var(--font-size-m);
        font-style: italic;
      }
    </style>
    <p class="blank"><slot></slot></p>
  `;
}

customElements.define('yzdt-blank-slate', component(blankSlate));
