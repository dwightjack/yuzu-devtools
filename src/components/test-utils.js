export const mockComponent = (tag) => (props) => {
  const el = document.createElement(tag);
  Object.keys(props).forEach((key) => {
    el.setAttribute(key, props[key]);
  });
  return el;
};
