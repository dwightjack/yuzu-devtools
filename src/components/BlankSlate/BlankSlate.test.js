import { toHTML } from '../test-utils';
import BlankSlate from './BlankSlate';

describe('BlankSlate', () => {
  test('default snapshot', () => {
    const { html } = toHTML(BlankSlate);
    expect(html).toMatchSnapshot();
  });
});
