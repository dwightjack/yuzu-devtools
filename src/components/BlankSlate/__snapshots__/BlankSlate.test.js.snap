// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankSlate default snapshot 1`] = `
<p class="blank">
  <slot>
  </slot>
</p>
`;
